src/14_legendre.jl

```diff
@@ -0,0 +1,370 @@
+        p = (2k-1)/(k-m)*x*p_prev - (k+m-1)/(k-m)*p_prev_prev       # = P_k^m(x)
+    end                                             # on output: p = P_n^m(x)
+
+    p
+end
+
+@doc raw"""
+    legendreq(n[, m], x)
+
+Evaluate the Legendre function of second kind ``Q_n(x)`` of integer degree ``n`` at position
+``x``.
+If the order ``m`` is supplied then the associated Legendre function of second kind
+``Q_n^{(m)}(x)`` is computed.
+Their definitions are given by
+
+```math
+\begin{align}
+    Q_n^{(m)}(x)
+    &=
+    (-1)^m (1-x^2)^\frac{m}{2} \frac{\mathrm{d}^m}{\mathrm{d}x^m} \left( Q_n(x) \right)
+    \,, \quad m \in \mathbb{N}_0 \,, \; x \in (-1,1)
+    \\
+    Q_n(x)
+    &=
+    \begin{cases}
+        \frac{1}{2} \log \frac{1+x}{1-x}
+        & \text{for} \quad n = 0
+        \\
+        P_1(x) Q_0(x) - 1
+        & \text{for} \quad n = 1
+        \\
+        \frac{2n-1}{n} x Q_{n-1}(x) - \frac{n-1}{n} Q_{n-2}(x)
+        & \text{for} \quad n \geq 2
+        \,.
+    \end{cases}
+\end{align}
+```
+where ``P_1`` is the Legendre polynomial of first kind and degree one.
+The functions diverge at the interval boundaries
+``\lim_{x\rightarrow \pm 1} |Q_n(x)| = \infty``.
+
+External links: [DLMF - Legendre function](https://dlmf.nist.gov/14.7.E2),
+[DLMF - associated Legendre function](https://dlmf.nist.gov/14.7.E9),
+[Wikipedia - Legendre function](https://en.wikipedia.org/wiki/Legendre_functions#Legendre_functions_of_the_second_kind_(Qn)),
+[Wikipedia - associated Legendre function](https://en.wikipedia.org/wiki/Legendre_function#Associated_Legendre_functions_of_the_second_kind).
+"""
+function legendreq(n::Integer, x)
+    (0 > n)      && throw(DomainError(n, "must be non-negative"))
+    (1 < abs(x)) && throw(DomainError(x, "must be in the range [-1,1]"))
+
+    if x == 1
+        return Inf
+    elseif x == -1
+        return (-1)^(n+1) * Inf
+    end
+
+    Q0 = 0.5 * log((1+x)/(1-x))
+    Q1 = legendrep(1, x) * Q0 - 1
+
+    ABC_recurrence(n, x,
+        m->(2m-1)//m, 0, m->(m-1)//m,       # A_m, B_m, C_m
+        Q0, Q1)
+end
+function legendreq(n::Integer, m::Integer, x)
+    (n <  0) && throw(DomainError(n, "must be non-negative"))
+    (m <  0) && throw(DomainError(m, "must be non-negative"))
+    (m == 0) && return legendreq(n, x)
+
+    # x check after: Q^0_n is also implemented for x = +-1
+    (1 <= abs(x)) && throw(DomainError(x, "must be in the range (-1,1)"))
+
+    # step 1: compute Q_n^0(x)
+    q_n_0 = legendreq(n, x)
+
+    # step 2: compute Q_n^1(x)
+    Q01 = -(1-x^2)^(-0.5)                                                   # = Q_0^1(x)
+    Q11 = -sqrt(1-x^2) * (0.5*log((1+x)/(1-x)) + x/(1-x^2))                 # = Q_1^1(x)
+    if n == 0
+        q = Q01
+    elseif n == 1
+        q = Q11
+    else
+        q, q_prev = Q11, Q01
+        for k = 2:n
+            q_prev, q_prev_prev = q, q_prev
+            q = (2k-1)//(k-1) * x * q_prev - k//(k-1) * q_prev_prev         # = Q_k^1(x)
```

Comment:
Rather than using rationals here, which are slow, better to use expressions like:
```jl
(2k-1) * x * q_prev / (k-1) - k*q_prev_prev/(k-1)
```
which will automatically do the division in the precision of `x`.

(Even just *constructing* a rational like `(2k-1)//(k-1)` is relatively slow because it will first run the [gcd algorithm](https://en.wikipedia.org/wiki/Euclidean_algorithm) to reduce the fraction to lowest terms.)

Reply:
converted rationals to floating divisions in commit [d7307f4](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/d7307f4fff2da592ad66e4acdcb9c403c91abbb9).